azavea/sentinel-2-stac-utils
# Parse CSV instead of using dummy data

## Overview

This PR uses [`kantan.csv`](https://nrinaudo.github.io/kantan.csv/), a typeclassy CSV library with refined and generic support, to read the csv passed in via the CLI from a local path or S3 instead of using dummy data.

### Checklist

- ~New tests have been added or existing tests have been modified~ lol someday :(

### Notes

I pulled example CSVs from what I generated in #14 to replace the original sample data

### Testing Instructions

- Run an l1c import locally: `bloop run s2stac -- create --collection L2A --inventory-path ./data/short-product-info-l2a.csv --output-catalog-root test/from-csv` (you can kill it pretty early, since we're proving CSV parsing, not the crawler)
- Run an l2a import against an s3 csv: `bloop run s2stac -- create --collection L2A --inventory-path s3://rasterfoundry-development-data-us-east-1/jisantuc-s2stactesting/queryable-inventory-batch-test-1-l2a/2020/07/25/data.csv --output-catalog-root test/from-csv-s3` (you can kill it pretty early, since we're proving CSV parsing, not the crawler)
- add a bogus line to the top of the local csv (it can be whatever you want, as long as it has too few fields, too many fields (?), or a field of the wrong type)
- try to run the import again -- you should see an error message about the value it didn't like (you can kill it pretty early since the error is in the first row)

Closes azavea/noaa-flood-mapping#16
